include/GafferSceneUI/LightPositionTool.h

```diff
@@ -0,0 +1,137 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2023, Cinesite VFX Ltd. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "GafferSceneUI/Export.h"
+#include "GafferSceneUI/TransformTool.h"
+#include "GafferSceneUI/TypeIds.h"
+
+#include "GafferUI/Handle.h"
+
+#include "Gaffer/ScriptNode.h"
+
+namespace GafferSceneUI
+{
+
+IE_CORE_FORWARDDECLARE( SceneView );
+
+class GAFFERSCENEUI_API LightPositionTool : public GafferSceneUI::TransformTool
+{
+
+	public :
+
+		LightPositionTool( SceneView *view, const std::string &name = defaultName<LightPositionTool>() );
+		~LightPositionTool() override;
+
+		GAFFER_NODE_DECLARE_TYPE( GafferSceneUI::LightPositionTool, LightPositionToolTypeId, TransformTool );
+
+		void position( const Imath::V3f &shadowPivot, const Imath::V3f &shadowTarget, const float targetDistance );
+
+	protected :
+
+		bool affectsHandles( const Gaffer::Plug *input ) const override;
+		void updateHandles( float rasterScale ) override;
+
+	private :
```

Comment:
I think just a `placementMode()` plug, which is a method rather than member data. Will that count as an ABI break? I could add it now and have it be a no-op until the specular mode is in place.

Otherwise I think the specular mode will have everything it needs - a target, the normals come from the buffer, viewer position comes from the viewer, and I can reuse what I'm now calling `DistanceHandle` (the handle formerly known as `ShadowHandle`).

Reply:
New plugs aren't an ABI break, so sounds like all is well.